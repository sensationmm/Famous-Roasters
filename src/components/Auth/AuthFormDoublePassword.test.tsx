import { fireEvent, render, waitFor } from '@testing-library/react'
import Form from 'rc-field-form'
import React from 'react'

import { AuthFormDoublePassword } from '.'

describe('Email and double Password fields group custom auth component', () => {
  const snippet = () => (
    <Form name="testAuthFormItemInput" method="POST">
      <AuthFormDoublePassword screenKey="signIn" onChange={() => null} />
    </Form>
  )

  it('Renders correctly', async () => {
    const { container } = render(snippet())
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('Password is too short error', async () => {
    const { container, getByTestId } = render(snippet())
    await waitFor(() => {
      const password = getByTestId('password')
      expect(password).toBeInTheDocument()
      fireEvent.click(password)
      fireEvent.change(password, { target: { value: '1234567' } })
      fireEvent.click(password)
    })
    expect(container).toMatchSnapshot()
  })

  it('Password is too long error', async () => {
    const { container, getByTestId } = render(snippet())
    await waitFor(() => {
      const password = getByTestId('password')
      expect(password).toBeInTheDocument()
      fireEvent.click(password)
      fireEvent.change(password, {
        target: {
          value:
            '12345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890',
        },
      })
      fireEvent.click(password)
    })
    expect(container).toMatchSnapshot()
  })

  it('Password must contain an uppercase error', async () => {
    const { container, getByTestId } = render(snippet())
    await waitFor(() => {
      const password = getByTestId('password')
      expect(password).toBeInTheDocument()
      fireEvent.click(password)
      fireEvent.change(password, { target: { value: 'abc1234@' } })
      fireEvent.click(password)
    })
    expect(container).toMatchSnapshot()
  })

  it('Password must contain a lowercase error', async () => {
    const { container, getByTestId } = render(snippet())
    await waitFor(() => {
      const password = getByTestId('password')
      expect(password).toBeInTheDocument()
      fireEvent.click(password)
      fireEvent.change(password, { target: { value: 'ABC1234@' } })
      fireEvent.click(password)
    })
    expect(container).toMatchSnapshot()
  })

  it('Password must contain a number error', async () => {
    const { container, getByTestId } = render(snippet())
    await waitFor(() => {
      const password = getByTestId('password')
      expect(password).toBeInTheDocument()
      fireEvent.click(password)
      fireEvent.change(password, { target: { value: 'AaAaAaA@' } })
      fireEvent.click(password)
    })
    expect(container).toMatchSnapshot()
  })

  it('Password must contain a special char error', async () => {
    const { container, getByTestId } = render(snippet())
    await waitFor(() => {
      const password = getByTestId('password')
      expect(password).toBeInTheDocument()
      fireEvent.click(password)
      fireEvent.change(password, { target: { value: 'Aa123456' } })
      fireEvent.click(password)
    })
    expect(container).toMatchSnapshot()
  })

  it('Password cannot contain spaces error', async () => {
    const { container, getByTestId } = render(snippet())
    await waitFor(() => {
      const password = getByTestId('password')
      expect(password).toBeInTheDocument()
      fireEvent.click(password)
      fireEvent.change(password, { target: { value: 'Aa 1234@' } })
      fireEvent.click(password)
    })
    expect(container).toMatchSnapshot()
  })

  it('Matching passwords is valid', async () => {
    const { container, getByTestId } = render(snippet())
    await waitFor(() => {
      const password = getByTestId('password')
      expect(password).toBeInTheDocument()
      fireEvent.click(password)
      fireEvent.change(password, { target: { value: 'Aa123qwertz@' } })
      fireEvent.click(password)
      const passwordRepeat = getByTestId('passwordRepeat')
      expect(passwordRepeat).toBeInTheDocument()
      fireEvent.click(passwordRepeat)
      fireEvent.change(passwordRepeat, { target: { value: 'Aa123qwertz@' } })
      fireEvent.click(passwordRepeat)
    })
    expect(container).toMatchSnapshot()
  })

  it('Matching passwords is invalid', async () => {
    const { container, getByTestId } = render(snippet())
    await waitFor(() => {
      const password = getByTestId('password')
      expect(password).toBeInTheDocument()
      fireEvent.click(password)
      fireEvent.change(password, { target: { value: 'Aa123qwertz@' } })
      fireEvent.click(password)
      const passwordRepeat = getByTestId('passwordRepeat')
      expect(passwordRepeat).toBeInTheDocument()
      fireEvent.click(passwordRepeat)
      fireEvent.change(passwordRepeat, { target: { value: 'notmatching' } })
      fireEvent.click(passwordRepeat)
    })
    expect(container).toMatchSnapshot()
  })
})
