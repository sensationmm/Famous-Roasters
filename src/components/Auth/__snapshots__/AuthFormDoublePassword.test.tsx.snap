// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Email and double Password fields group custom auth component Matching passwords is invalid 1`] = `
<div>
  <form
    method="POST"
  >
    <div>
      <label
        class="relative"
      >
        <span
          class="absolute -top-6 left-3 px-1 bg-white"
        >
          <span
            class="text-sm leading-5"
          >
            Dein Passwort
          </span>
        </span>
        <input
          class="w-full border rounded-full focus:outline-0 focus:ring px-4 py-2 focus:ring-primary/30"
          data-testid="password"
          placeholder="Dein Passwort"
          type="password"
          value="Aa123qwertz@"
        />
      </label>
    </div>
    <div
      class="mt-8"
    >
      <label
        class="relative"
      >
        <span
          class="absolute -top-6 left-3 px-1 bg-white"
        >
          <span
            class="text-sm leading-5"
          >
            auth.signIn.passwordRepeat.label
          </span>
        </span>
        <input
          class="w-full border rounded-full focus:outline-0 focus:ring px-4 py-2 focus:ring-primary/30"
          data-testid="passwordRepeat"
          placeholder="auth.signIn.passwordRepeat.placeholder"
          type="password"
          value="notmatching"
        />
      </label>
    </div>
  </form>
</div>
`;

exports[`Email and double Password fields group custom auth component Matching passwords is valid 1`] = `
<div>
  <form
    method="POST"
  >
    <div>
      <label
        class="relative"
      >
        <span
          class="absolute -top-6 left-3 px-1 bg-white"
        >
          <span
            class="text-sm leading-5"
          >
            Dein Passwort
          </span>
        </span>
        <input
          class="w-full border rounded-full focus:outline-0 focus:ring px-4 py-2 focus:ring-primary/30"
          data-testid="password"
          placeholder="Dein Passwort"
          type="password"
          value="Aa123qwertz@"
        />
      </label>
    </div>
    <div
      class="mt-8"
    >
      <label
        class="relative"
      >
        <span
          class="absolute -top-6 left-3 px-1 bg-white"
        >
          <span
            class="text-sm leading-5"
          >
            auth.signIn.passwordRepeat.label
          </span>
        </span>
        <input
          class="w-full border rounded-full focus:outline-0 focus:ring px-4 py-2 focus:ring-primary/30"
          data-testid="passwordRepeat"
          placeholder="auth.signIn.passwordRepeat.placeholder"
          type="password"
          value="Aa123qwertz@"
        />
      </label>
    </div>
  </form>
</div>
`;

exports[`Email and double Password fields group custom auth component Password cannot contain spaces error 1`] = `
<div>
  <form
    method="POST"
  >
    <div>
      <label
        class="relative"
      >
        <span
          class="absolute -top-6 left-3 px-1 bg-white"
        >
          <span
            class="text-sm leading-5"
          >
            Dein Passwort
          </span>
        </span>
        <input
          class="w-full border rounded-full focus:outline-0 focus:ring px-4 py-2 focus:ring-primary/30"
          data-testid="password"
          placeholder="Dein Passwort"
          type="password"
          value="Aa 1234@"
        />
      </label>
    </div>
    <div
      class="mt-8"
    >
      <label
        class="relative"
      >
        <span
          class="absolute -top-6 left-3 px-1 bg-white"
        >
          <span
            class="text-sm leading-5"
          >
            auth.signIn.passwordRepeat.label
          </span>
        </span>
        <input
          class="w-full border rounded-full focus:outline-0 focus:ring px-4 py-2 focus:ring-primary/30"
          data-testid="passwordRepeat"
          placeholder="auth.signIn.passwordRepeat.placeholder"
          type="password"
          value=""
        />
      </label>
    </div>
  </form>
</div>
`;

exports[`Email and double Password fields group custom auth component Password is too long error 1`] = `
<div>
  <form
    method="POST"
  >
    <div>
      <label
        class="relative"
      >
        <span
          class="absolute -top-6 left-3 px-1 bg-white"
        >
          <span
            class="text-sm leading-5"
          >
            Dein Passwort
          </span>
        </span>
        <input
          class="w-full border rounded-full focus:outline-0 focus:ring px-4 py-2 focus:ring-primary/30"
          data-testid="password"
          placeholder="Dein Passwort"
          type="password"
          value="1234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890"
        />
      </label>
    </div>
    <div
      class="mt-8"
    >
      <label
        class="relative"
      >
        <span
          class="absolute -top-6 left-3 px-1 bg-white"
        >
          <span
            class="text-sm leading-5"
          >
            auth.signIn.passwordRepeat.label
          </span>
        </span>
        <input
          class="w-full border rounded-full focus:outline-0 focus:ring px-4 py-2 focus:ring-primary/30"
          data-testid="passwordRepeat"
          placeholder="auth.signIn.passwordRepeat.placeholder"
          type="password"
          value=""
        />
      </label>
    </div>
  </form>
</div>
`;

exports[`Email and double Password fields group custom auth component Password is too short error 1`] = `
<div>
  <form
    method="POST"
  >
    <div>
      <label
        class="relative"
      >
        <span
          class="absolute -top-6 left-3 px-1 bg-white"
        >
          <span
            class="text-sm leading-5"
          >
            Dein Passwort
          </span>
        </span>
        <input
          class="w-full border rounded-full focus:outline-0 focus:ring px-4 py-2 focus:ring-primary/30"
          data-testid="password"
          placeholder="Dein Passwort"
          type="password"
          value="1234567"
        />
      </label>
    </div>
    <div
      class="mt-8"
    >
      <label
        class="relative"
      >
        <span
          class="absolute -top-6 left-3 px-1 bg-white"
        >
          <span
            class="text-sm leading-5"
          >
            auth.signIn.passwordRepeat.label
          </span>
        </span>
        <input
          class="w-full border rounded-full focus:outline-0 focus:ring px-4 py-2 focus:ring-primary/30"
          data-testid="passwordRepeat"
          placeholder="auth.signIn.passwordRepeat.placeholder"
          type="password"
          value=""
        />
      </label>
    </div>
  </form>
</div>
`;

exports[`Email and double Password fields group custom auth component Password must contain a lowercase error 1`] = `
<div>
  <form
    method="POST"
  >
    <div>
      <label
        class="relative"
      >
        <span
          class="absolute -top-6 left-3 px-1 bg-white"
        >
          <span
            class="text-sm leading-5"
          >
            Dein Passwort
          </span>
        </span>
        <input
          class="w-full border rounded-full focus:outline-0 focus:ring px-4 py-2 focus:ring-primary/30"
          data-testid="password"
          placeholder="Dein Passwort"
          type="password"
          value="ABC1234@"
        />
      </label>
    </div>
    <div
      class="mt-8"
    >
      <label
        class="relative"
      >
        <span
          class="absolute -top-6 left-3 px-1 bg-white"
        >
          <span
            class="text-sm leading-5"
          >
            auth.signIn.passwordRepeat.label
          </span>
        </span>
        <input
          class="w-full border rounded-full focus:outline-0 focus:ring px-4 py-2 focus:ring-primary/30"
          data-testid="passwordRepeat"
          placeholder="auth.signIn.passwordRepeat.placeholder"
          type="password"
          value=""
        />
      </label>
    </div>
  </form>
</div>
`;

exports[`Email and double Password fields group custom auth component Password must contain a number error 1`] = `
<div>
  <form
    method="POST"
  >
    <div>
      <label
        class="relative"
      >
        <span
          class="absolute -top-6 left-3 px-1 bg-white"
        >
          <span
            class="text-sm leading-5"
          >
            Dein Passwort
          </span>
        </span>
        <input
          class="w-full border rounded-full focus:outline-0 focus:ring px-4 py-2 focus:ring-primary/30"
          data-testid="password"
          placeholder="Dein Passwort"
          type="password"
          value="AaAaAaA@"
        />
      </label>
    </div>
    <div
      class="mt-8"
    >
      <label
        class="relative"
      >
        <span
          class="absolute -top-6 left-3 px-1 bg-white"
        >
          <span
            class="text-sm leading-5"
          >
            auth.signIn.passwordRepeat.label
          </span>
        </span>
        <input
          class="w-full border rounded-full focus:outline-0 focus:ring px-4 py-2 focus:ring-primary/30"
          data-testid="passwordRepeat"
          placeholder="auth.signIn.passwordRepeat.placeholder"
          type="password"
          value=""
        />
      </label>
    </div>
  </form>
</div>
`;

exports[`Email and double Password fields group custom auth component Password must contain a special char error 1`] = `
<div>
  <form
    method="POST"
  >
    <div>
      <label
        class="relative"
      >
        <span
          class="absolute -top-6 left-3 px-1 bg-white"
        >
          <span
            class="text-sm leading-5"
          >
            Dein Passwort
          </span>
        </span>
        <input
          class="w-full border rounded-full focus:outline-0 focus:ring px-4 py-2 focus:ring-primary/30"
          data-testid="password"
          placeholder="Dein Passwort"
          type="password"
          value="Aa123456"
        />
      </label>
    </div>
    <div
      class="mt-8"
    >
      <label
        class="relative"
      >
        <span
          class="absolute -top-6 left-3 px-1 bg-white"
        >
          <span
            class="text-sm leading-5"
          >
            auth.signIn.passwordRepeat.label
          </span>
        </span>
        <input
          class="w-full border rounded-full focus:outline-0 focus:ring px-4 py-2 focus:ring-primary/30"
          data-testid="passwordRepeat"
          placeholder="auth.signIn.passwordRepeat.placeholder"
          type="password"
          value=""
        />
      </label>
    </div>
  </form>
</div>
`;

exports[`Email and double Password fields group custom auth component Password must contain an uppercase error 1`] = `
<div>
  <form
    method="POST"
  >
    <div>
      <label
        class="relative"
      >
        <span
          class="absolute -top-6 left-3 px-1 bg-white"
        >
          <span
            class="text-sm leading-5"
          >
            Dein Passwort
          </span>
        </span>
        <input
          class="w-full border rounded-full focus:outline-0 focus:ring px-4 py-2 focus:ring-primary/30"
          data-testid="password"
          placeholder="Dein Passwort"
          type="password"
          value="abc1234@"
        />
      </label>
    </div>
    <div
      class="mt-8"
    >
      <label
        class="relative"
      >
        <span
          class="absolute -top-6 left-3 px-1 bg-white"
        >
          <span
            class="text-sm leading-5"
          >
            auth.signIn.passwordRepeat.label
          </span>
        </span>
        <input
          class="w-full border rounded-full focus:outline-0 focus:ring px-4 py-2 focus:ring-primary/30"
          data-testid="passwordRepeat"
          placeholder="auth.signIn.passwordRepeat.placeholder"
          type="password"
          value=""
        />
      </label>
    </div>
  </form>
</div>
`;

exports[`Email and double Password fields group custom auth component Renders correctly 1`] = `
<div>
  <form
    method="POST"
  >
    <div>
      <label
        class="relative"
      >
        <span
          class="absolute -top-6 left-3 px-1 bg-white"
        >
          <span
            class="text-sm leading-5"
          >
            Dein Passwort
          </span>
        </span>
        <input
          class="w-full border rounded-full focus:outline-0 focus:ring px-4 py-2 focus:ring-primary/30"
          data-testid="password"
          placeholder="Dein Passwort"
          type="password"
          value=""
        />
      </label>
    </div>
    <div
      class="mt-8"
    >
      <label
        class="relative"
      >
        <span
          class="absolute -top-6 left-3 px-1 bg-white"
        >
          <span
            class="text-sm leading-5"
          >
            auth.signIn.passwordRepeat.label
          </span>
        </span>
        <input
          class="w-full border rounded-full focus:outline-0 focus:ring px-4 py-2 focus:ring-primary/30"
          data-testid="passwordRepeat"
          placeholder="auth.signIn.passwordRepeat.placeholder"
          type="password"
          value=""
        />
      </label>
    </div>
  </form>
</div>
`;
